### PR TITLE
brigade-cli 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by krancour**
Installs the CLI component of Brigade (brig).
Server side components are installed separately using Helm.

Signed-off-by: Kent Rancourt <kent.rancourt@microsoft.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/brigade-cli.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["SKIP_DOCKER"] = "true"
+    ENV["VERSION"] = "v#{version}"
+
+    system "make", "hack-build-cli"
+
+    os = `go env GOOS`.strip
+    arch = `go env GOARCH`.strip
```
Comment (carlocab) [inline Formula/brigade-cli.rb:19]:
```suggestion
    os = Utils.safe_popen_read("go", "env", "GOOS").strip
    arch = Utils.safe_popen_read("go", "env", "GOARCH").strip
```

File: Formula/brigade-cli.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  test do
+    system bin/"brig", "init", "--id", "foo"
+    assert File.directory? testpath/".brigade"
+
+    version_output = shell_output(bin/"brig version 2>&1")
+    assert_match "Brigade client:", version_output
+    if build.stable?
+      commit = stable.instance_variable_get(:@resource).instance_variable_get(:@specs)[:revision][0..6]
+      assert_match "Brigade client: version v#{version} -- commit #{commit}", version_output
+    end
```
Comment (carlocab) [inline Formula/brigade-cli.rb:32]:
```suggestion
    return unless stable?

    commit = stable.specs[:revision][0..6]
    assert_match "Brigade client: version v#{version} -- commit #{commit}", version_output
```
I'm not a fan of using `stable.specs` here -- I'm pretty sure that's a private API.

File: Formula/brigade-cli.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    system "make", "hack-build-cli"
+
+    os = `go env GOOS`.strip
+    arch = `go env GOARCH`.strip
+    bin.install "bin/brig-#{os}-#{arch}" => "brig"
+  end
+
+  test do
+    system bin/"brig", "init", "--id", "foo"
+    assert File.directory? testpath/".brigade"
```
Comment (carlocab) [inline Formula/brigade-cli.rb:25]:
```suggestion
    assert_predicate testpath/".brigade", :directory?
```

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @krancour. A few suggestions.

**Comment by krancour** [comment]
@carlocab thanks for the timely review!

I have applied all your feedback and force-pushed.

**Review by SMillerDev — APPROVED** [review]
Thanks @krancour  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**